app/src/main/java/com/keren/articles/customComponents/Resource.kt

```diff
@@ -0,0 +1,7 @@
+package com.keren.articles.customComponents
+
+sealed class Resource<out T> {
+    data class Success<out T>(val data: T? = null) : Resource<T>()
+    class NetworkError<out T>(val error: String) : Resource<T>()
+    class Loading<out T> : Resource<T>()
+}
```

Comment:
Some improvements:
```suggestion
sealed interface Resource<out T> {
    data class Success<out T>(val data: T? = null) : Resource<T>
    data class Error(val error: String) : Resource<Nothing>
    object Loading : Resource<Nothing>
}
```

Please make sure you understand what `out` is